### PR TITLE
Init CRUD template for comment table and get comment request

### PR DESCRIPTION
**Pull request by jinchen1036**
### Overview of PR
#77 
- [x] Create a GET request `{url}/api/v1/post/viewcomments`, returns all the comments of the requested post from the database
- [x] Should support query string `{url}/api/v1/post/viewcomments? postID=a&currentPage=b&pageSize=c`
  - `postID` is required
  - `currentPage`  is optional, **_index starts with 1_**, default to 1 if absence
  - `pageSize`  is optional, default to 10 if absence
- [x] Unit Testing

### Technical Changes
- Create **post** folder in `app`
  - Have `index.js` and `getComments.js` with corresponding unit tests
- Create `commentService.js` in `services` folder
  - With CRD comment table template

### Architecture Changes
- Added new route `{{localhost}}/api/v1/post/viewcomments`

### CODE REVIEW

File: src/app/post/index.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const viewCommentValidation = validateRequest([
+  check('postID', 'Invalid postID')
+    .isNumeric()
+    .not().isEmpty(),
+  check('pageSize', 'Invalid pageSize number')
+    .optional()
+    .isNumeric(),
+  check('currentPage', 'Invalid currentPage number')
+    .optional()
+    .isNumeric(),
```
Comment (JiejayLan) [inline src/app/post/index.js]: should be an int

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Logs
 logs
 *.log
+*.lock
```
Comment (JiejayLan) [inline .gitignore]: what is '*.lock' used for?

Comment (jinchen1036) [reply .gitignore]: For ignore yarn.lock file

File: src/app/post/index.js (hunk 1 above)
Comment (jinchen1036) [reply src/app/post/index.js]: Fixed

File: .gitignore (hunk 2 above)
Comment (songrenzhao) [reply .gitignore]: general speaking, we should not ignore yarn.lock and package-lock.json. Please see [this](https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/)

Please remove `yarn.lock`, we already agreed not to use yarn.

Comment (jinchen1036) [reply .gitignore]: Add the yarn.lock back.

**Pull request merged**